backend.native/tests/runtime/workers/lazy1.kt

```diff
@@ -13,10 +13,43 @@ class Leak {
     val leak by lazy { this }
 }
 
-@Test fun runTest() {
+class SelfReference {
+    val x = 17
+    val self by lazy { this }
+    val recursion: Int by lazy {
+        if (x < 17) 42 else recursion
+    }
+    val freezer: Int by lazy {
+        freeze()
+        42
+    }
+}
+
+
+
+@Test fun runTest1() {
+    assertFailsWith<IllegalStateException> {
+        println(SelfReference().recursion)
+    }
+    assertFailsWith<IllegalStateException> {
+        println(SelfReference().freeze().recursion)
+    }
+}
+
+// Disabled, as requires cyclic collector.
```

Comment:
I'm afraid this PR would do more harm than good.
I expect number of users having memory leaks in 1.3.70 because of it. Not sure if users have encountered the incorrect behaviour too often.

Reply:
They would have leaks only in very convoluted cases which didn't work before (throwing an exception).

Reply:
I see.